From ddcfacb7dd10429ff5e57845d18657c8f1dc2997 Mon Sep 17 00:00:00 2001
From: 7980062 <ken.nicolson@jp.panasonic.com>
Date: Fri, 15 Feb 2019 12:41:01 +0900
Subject: [PATCH] Hack to get around GitHub playing with TAR

For whatever reason, GitHub randomly removes the version string from
the root directory of downloaded TAR files.

Try to address this by having both a versioned and an unversioned name
for the root directory, and if either match, accept that.

A fuller solution might add a "unversioned" parameter to checkout() so
we can turn this feature on and off, but that is a task for another day.
---
 src/vcstools/tar.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/vcstools/tar.py b/src/vcstools/tar.py
index d27df9e..9e90cf0 100644
--- a/src/vcstools/tar.py
+++ b/src/vcstools/tar.py
@@ -46,6 +46,7 @@
 import tarfile
 import sys
 import yaml
+import re
 from vcstools.vcs_base import VcsClientBase, VcsError
 from vcstools.common import urlretrieve_netrc, ensure_dir_notexists
 
@@ -117,8 +118,9 @@ def checkout(self, url, version='', verbose=False,
                 # relative path
                 subdirs = []
                 members = []
+                unversioned = re.sub(r'-[0-9.-]+$', '', version)
                 for m in temp_tarfile.getmembers():
-                    if m.name.startswith(version + '/'):
+                    if m.name.startswith(version + '/') or m.name.startswith(unversioned + '/'):
                         members.append(m)
                     if m.name.split('/')[0] not in subdirs:
                         subdirs.append(m.name.split('/')[0])
